### PR TITLE
[WIP] Migration to Hyper v0.12

### PR DESCRIPTION
**Pull request by whitfin**
This PR includes migration to the new version of Tokio, via the `tokio` crate. It includes an update to Hyper 0.12, seeing as that's required for Hyper to function with Tokio correctly.

### CODE REVIEW

File: gotham/src/os/mod.rs [hunk 1]
```diff
@@ -1,9 +0,0 @@
-#[cfg(not(windows))]
```
Comment (nyarly) [inline gotham/src/os/mod.rs]: This makes me super happy.

File: gotham/src/test/mod.rs [hunk 2]
```diff
@@ -173,13 +171,19 @@ where
...
 
-        handle.spawn(f);
+        {
+            self.data.runtime.read().unwrap().spawn(f);
+        }
 
-        let client = Client::configure()
-            .connector(TestConnect {
-                stream: cell::RefCell::new(Some(cs)),
+        let client = Core::new()
```
Comment (nyarly) [inline gotham/src/test/mod.rs]: This looks like the crucial remaining usage of tokio_core, right?

Comment (whitfin) [reply gotham/src/test/mod.rs]: @nyarly yep! This is the one that has to stick around until Hyper 0.12 is adopted.

**Comment by whitfin** [comment]
This is getting there very slowly; FWIW once it actually compiles, we should go over it to clean up before merging - I'm very much taking a quick approach to just get something working for now.

**Comment by whitfin** [comment]
Closing in favour of #246.

**Pull request closed without merging**